cpp/src/io/orc/orc.cpp

```diff
@@ -182,6 +182,9 @@ void ProtobufReader::read(timestamp_statistics& s, size_t maxlen)
                        field_reader(5, s.minimum_nanos),
                        field_reader(6, s.maximum_nanos));
   function_builder(s, maxlen, op);
+  // adjust nanoseconds because someone had a bright idea to store nanoseconds + 1 in the protobuf
+  if (s.minimum_nanos.has_value()) { --s.minimum_nanos.value(); }
+  if (s.maximum_nanos.has_value()) { --s.maximum_nanos.value(); }
```

Comment:
Since we are using unsigned int, will this underflow?

Reply:
The nanoseconds are stored as `value+1`, so zero is not a valid content here. We found out the hard way about this :D

Reply:
Can you add a comment to clarify that please? So we won't forget about it.

Reply:
Oh, approved but forgot about this :smile:

Reply:
no worries, I haven't ;)